### DISCUSSION TITLE
homebrew/cask/unison 2.51.5 "Unison.app” is damaged and can’t be opened. You should move it to the Trash."

### DISCUSSION THREAD

**Discussion by reagle** [Homebrew/discussions#2672] [Casks]
I believe I got an update yesterday to the unison cask and can no longer launch it: `"Unison.app” is damaged and can’t be opened. You should move it to the Trash.`

I uninstalled it and installed the brew version of 2.51.5 and it seems to be working, but don't have access to the GUI of the cask build.

Has this happened to anyone else?

**Comment by neutric** [comment]
I can reproduce this for the cask. This also happens when downloading the tar.gz manually and dragging the app bundle into `/Applications`. And it is not limited to this release. Same happens for the previous one, didn't check further back.

https://github.com/bcpierce00/unison/wiki/CI-Binary-instructions via https://github.com/bcpierce00/unison/issues/584 demands

> Before running it one must do `xattr -cr /Applications/Unison.app`; otherwise, an unhelpful error message about a damaged app is produced.

`xattr -cr` blows away extended attributes. After that, you can start the GUI app no problem. But which extended attributes are the problem here?

```
xattr -lr /Applications/Unison.app/
/Applications/Unison.app//Contents/_CodeSignature/CodeResources: com.apple.quarantine: 0181;61c3a87b;Homebrew\x20Cask;B1571C93-6D75-483D-B4E4-BC6CF53C7F77
/Applications/Unison.app//Contents/_CodeSignature: com.apple.quarantine: 0181;61c3a87b;Homebrew\x20Cask;B1571C93-6D75-483D-B4E4-BC6CF53C7F77
/Applications/Unison.app//Contents/MacOS/cltool: com.apple.metadata:kMDItemAlternateNames: ("unison")
/Applications/Unison.app//Contents/MacOS/cltool: com.apple.quarantine: 0181;61c3a87b;Homebrew\x20Cask;B1571C93-6D75-483D-B4E4-BC6CF53C7F77
[...]
```

The OS put this app in quarantine. Installing with `brew install --no-quarantine --cask unison` circumvents this security feature. For more info, https://github.com/Homebrew/homebrew-cask/blob/HEAD/doc/faq/app_cant_be_opened.md is recommended.

Feel free to ask further questions. If satisfied, please mark this as answered.

**Comment by reagle** [comment]
@neutric, thanks! 

1. The link to the homebrew-cask FAQ doesn't work. 
2. Do I understand this is something users must always do? This isn't something that can be fixed?
3. If I've already installed with `--no-quarantine`, when the cask is updates, does that option carry forward? I suspect not but suggest it should.

**Reply by SMillerDev** [reply]
See https://docs.brew.sh/FAQ#why-cant-i-open-a-mac-app-from-an-unidentified-developer instead, the PR moving that file was merged an hour after posting this.

**Comment by reagle** [comment]
@SMillerDev thanks. I'm still wondering about updates thought. *If* brew `no-quarantine` is not persistent on updates, it'd be handy if it gave a warning in the console to remind the user.
